docs/chapters/ci-and-cd/setting-up-automated-testing-yocto.rst

```diff
@@ -0,0 +1,176 @@
+
+Please note that the last argument to mkefidisk.sh is the name of the device on the target HW, e.g. /dev/sda or /dev/mmcblk2, etc.
+
+.. code-block:: bash
+
+    dd if=/dev/zero of=output.img count=1024 bs=4M
+    sudo losetup --find --show output.img
+    sudo ../sources/poky/scripts/contrib/mkefidisk.sh /dev/loop<NUMBER FROM PREVIOUS COMMAND> tmp/deploy/images/intel-corei7-64/core-image-testmaster-intel-corei7-64.hddimg /dev/sda
+    sudo losetup -d /dev/loop<NUMBER FROM PREVIOUS COMMAND>
+
+We now have a 4GB big disk image which we can write to a USB-stick.
+
+.. code-block:: bash
+
+    sudo dd if=output.img of=/dev/sd<YOUR USB DEVICE> bs=10M status=progress
+
+When this is completed, remove and re-insert the USB-stick. Partition #1-3 will be created now, but we need to manually add
+partition #4.
+
+.. code-block:: bash
+
+    sudo fdisk /dev/sd<YOUR USB DEVICE>
+    - Press n for new partition
+    - Press p for a primary partition
+    - Press enter twice to get it to fill all remaining space
+    - Press w to write
+    sudo mkfs.ext3 -L testrootfs /dev/sd<YOUR USB DEVICE>4
+
+Now mount the testmaster rootfs in order to do some changes there:
+
+.. code-block:: bash
+
+    sudo mount /dev/sd<YOUR USB DEVICE>2 /mnt/
+      
+    # Create image used to identify if system is booted into the testmaster image
+    sudo touch /mnt/etc/masterimage
+      
+    # Make sure the testmaster image shows a login prompt
+    sudo ln -sf /lib/systemd/system/getty@.service /mnt/etc/systemd/system/getty.target.wants/getty@tty1.service
+      
+    # Create a network conf file which we then copy to rootfs, this one uses DHCP
+    cat <<EOF > /tmp/20-wired.network
+    [Match]
+    Name=en*
+     
+    [Network]
+    DHCP=ipv4
+     
+    [DHCP]
+    RouteMetric=10
+    ClientIdentifier=mac
+    EOF
+    sudo cp /tmp/20-wired.network /mnt/etc/systemd/network/
+     
+      
+    # Unmount
+    sudo umount /mnt
+
+Now we mount the EFI partition to add a bootloader entry called "test" which boots the kernel and filesystem under test.
+
+.. code-block:: bash
+
+    # Set correct label on EFI partition
+    sudo dosfslabel /dev/sd<YOUR USB DEVICE>1 boot
+      
+    # Mount EFI partition
+    sudo mount /dev/sd<YOUR USB DEVICE>1 /mnt/
+    # Create temp bootloader config file which we then copy
+    cat <<EOF > /tmp/test.conf
+    title test
+    linux /test-kernel
+    options LABEL=test root=/dev/sda4 ro rootwait console=ttyS0 console=tty0
+    EOF
+    sudo cp /tmp/test.conf /mnt/loader/entries/
+      
+    # Unmount
+    sudo umount /mnt
+
+The USB-stick should now be ready and can be inserted into a NUC and booted, do that and check what IP-address it gets
+using e.g. "ip a".
+
+
+Building and testing an image
+-----------------------------
+Current issues in poky means we have to apply a patch needs to be applied: 
+`0001-Fix-automated-runtime-testing-using-SystemdbootTarge.patch` which is located in _TODO:_ Add link!
```

Comment:
could you perhaps add the patch in to the SoftwareFactory for now so people can find it and once we have it somewhere else we can point there instead?